### PR TITLE
feat: Better data validity propagation for return channels

### PR DESCRIPTION
**Pull request by phako**
- **fix: Ignore data validity through return channels**
- **chore: Minor linter fixes**

### CODE REVIEW

File: src/MetaDataPropagatingRegisterDecorator.cc [hunk 1]
```diff
@@ -21,13 +21,16 @@ namespace ChimeraTK {
     // Check if the data validity flag changed. If yes, propagate this information to the owning module and the application
     if(_dataValidity != _lastValidity) {
       if(_dataValidity == DataValidity::faulty) { // data validity changes to faulty
+        // Only propagate the validity to the owning module if we are not the return channel
```
Comment (mhier) [inline src/MetaDataPropagatingRegisterDecorator.cc:24]: Is this commit still correct?

File: src/MetaDataPropagatingRegisterDecorator.cc [hunk 2]
```diff
@@ -21,13 +21,16 @@ namespace ChimeraTK {
...
         _owner->incrementDataFaultCounter();
-        // external inpput in a circular dependency network
+        // external input in a circular dependency network
         if(_owner->getCircularNetworkHash() && !_isCircularInput) {
           ++(Application::getInstance()._circularNetworkInvalidityCounters[_owner->getCircularNetworkHash()]);
         }
       }
       else { // data validity changed to OK
+        // Only propagate the validity to the owning module if we are not the return channel
+
```
Comment (mhier) [inline src/MetaDataPropagatingRegisterDecorator.cc:33]: dito

File: tests/executables_src/testBidirectionalVariables.cc [hunk 3]
```diff
@@ -544,16 +544,20 @@ BOOST_AUTO_TEST_CASE(testDataValidityReturn) {
     app.lower.incrementDataFaultCounter();
     app.lower.var = 120;
     app.lower.var.write();
+    app.lower.decrementDataFaultCounter();
```
Comment (mhier) [inline tests/executables_src/testBidirectionalVariables.cc:547]: Just for clarification, I would move this line down below the two corresponding BOOST_CHECK lines, as it is not part of this test really but rather just restores the state as before the test.

**Review by mhier — APPROVED** [review]

**Review by phako — COMMENTED** [review]
Also added fixes for testPropagateDataFaultFlag which I forgot to push

**Comment by mhier** [comment]
go ahead with the merge when ready.

**Pull request merged**